pkg/kubectl/polymorphichelpers/logsforobject.go

```diff
@@ -68,10 +68,37 @@ func logsForObjectWithClient(clientset corev1client.CoreV1Interface, object, opt
 	case *corev1.Pod:
 		// if allContainers is true, then we're going to locate all containers and then iterate through them. At that point, "allContainers" is false
 		if !allContainers {
-			return []rest.ResponseWrapper{clientset.Pods(t.Namespace).GetLogs(t.Name, opts)}, nil
+			var container *v1.Container
+			if opts == nil || len(opts.Container) == 0 {
```

Comment:
This else-if keeps the `kubectl` behaviour the same to a user as in previous versions, but the output will be a bit different. For example, for `kubectl -n logging-test logs -lapp=logging_test` where `-lapp=logging_test` matches a pod with more than 1 container will be:

```
Error from server (BadRequest): a container name must be specified for pod logging-test-be-7657b857c7-4b9jr, choose one of: [logging-container-1 logging-container-2]
```

New output:

```
error: a container name must be specified for pod logging-test-be-7657b857c7-4b9jr, choose one of: [logging-container-1 logging-container-2]
```

Same when provided container name is incorrect (`kubectl -n logging-test logs -lapp=logging_test -c test`):

```
error: container test is not valid for pod logging-test-be-7657b857c7-4b9jr
```

instead of 

```
Error from server (BadRequest): container test is not valid for pod logging-test-be-7657b857c7-4b9jr
```

It happens because now we check pod object on the client before sending request to get logs.

Reply:
It works, but I'm not very happy about this approach for several reasons:

* On client I duplicate some of the logic we already have on server (see [`LogLocation ` from `pkg/registry/core/pod/strategy.go`](https://github.com/kubernetes/kubernetes/blob/18b4e1b84c241b89392a37721607d0b19382b92f/pkg/registry/core/pod/strategy.go#L286-L361))
* `LogsForObjectResponseWrapper` looks a bit clumsy

Any suggestions are very welcome.

Reply:
> On client I duplicate some of the logic we already have on server (see LogLocation from pkg/registry/core/pod/strategy.go)

ideally there should be a common backend.
although i'm not sure where it should be placed.

> LogsForObjectResponseWrapper looks a bit clumsy

possibly change the function name to better outline what it does.
e.g. SourceObjectResponseWrapper
can't think of good names ATM.

Reply:
> It works, but I'm not very happy about this approach for several reasons:

Ditto, I'd imagine this more like a writer wrapper in `kubectl logs` and on the backend I'd be inclined to propose to modify `LogsForObjectFunc` so that it returns `map[corev1.ObjectReference]rest.ResponseWrapper` instead of current `[]rest.ResponseWrapper` wdyt? The reason why I'm going with `corev1.ObjectReference` is that this does not restrict us into the format of the eventual string we'd put there and will allow consumers to format it per their own taste.